packages/txservice/src/transaction.ts

```diff
@@ -258,46 +201,111 @@ export class Transaction {
         // NOTE: This is the official receipt with status of 0, so it's safe to say the
         // transaction was in fact reverted and we should throw here.
         this.receipt = error.receipt;
+        this._validated = true;
+        this._error = error;
         throw error;
       } else {
+        this._error = error;
         throw error;
       }
     } else {
       this.receipt = result.value;
+      this._validated = true;
+      // Sanity checks.
+      if (this.receipt == null) {
+        // Receipt is undefined or null. This normally should never occur.
+        throw new TransactionServiceFailure("Unable to obtain receipt: ethers responded with null.", {
+          method,
+          receipt: this.receipt,
+          hash: this.response.hash,
+          id: this.id,
+        });
+      } else if (this.receipt.status === 0) {
+        // This should never occur. We should always get a TransactionReverted error in this event.
+        throw new TransactionServiceFailure("Transaction was reverted but TransactionReverted error was not thrown.", {
+          method,
+          receipt: this.receipt,
+          hash: this.response.hash,
+          id: this.id,
+        });
+      } else if (this.receipt.confirmations < 1) {
+        // Again, should never occur.
+        throw new TransactionServiceFailure("Receipt did not have any confirmations, should have timed out!", {
+          method,
+          receipt: this.receipt,
+          hash: this.response.hash,
+          id: this.id,
+        });
+      }
     }
+  }
 
-    // Sanity checks.
-    if (this.receipt == null) {
-      // Receipt is undefined or null. This normally should never occur.
-      throw new TransactionServiceFailure("Unable to obtain receipt: ethers responded with null.", {
-        method,
-        receipt: this.receipt,
-        hash: response.hash,
-        id: this.id,
-      });
-    } else if (this.receipt.status === 0) {
-      // This should never occur. We should always get a TransactionReverted error in this event.
-      throw new TransactionServiceFailure("Transaction was reverted but TransactionReverted error was not thrown.", {
+  /**
+   * Bump the gas price for this tx up by the configured percentage.
+   */
+  public bumpGasPrice() {
+    const currentPrice = this.gasPrice.get();
+    // Scale up gas by percentage as specified by config.
+    // TODO: Replace with actual config.
+    const bumpedGasPrice = currentPrice.add(currentPrice.mul(DEFAULT_CONFIG.gasReplacementBumpPercent).div(100)).add(1);
+    this.gasPrice.set(bumpedGasPrice);
+    this.logger.info(
+      {
+        method: this.bumpGasPrice.name,
+        previousGasPrice: currentPrice.toString(),
+        newGasPrice: bumpedGasPrice.toString(),
+      },
+      "Bumping tx gas price for reattempt.",
+    );
+  }
+
+  private timeUntilExpiry(): number {
+    const expiry = this.timestamp + this.provider.confirmationTimeout;
+    return expiry - Date.now();
```

Comment:
This could be negative, but we shouldn't let that prevent "attempting" to confirm right? Maybe have a minimum of 2-5 seconds just to let validate() attempt to validate